jsapp/js/searches.es6

```diff
@@ -117,9 +118,15 @@ function SearchContext(opts={}) {
       }
     },
     // remove asset from all search store lists
-    removeAsset(asset) {
-      // only update things if given asset matches the current context types
-      if (this.state.defaultQueryFilterParams?.assetType.includes(asset.assetType)) {
+    removeAsset(asset, isNonOwner) {
```

Comment:
If I get it right - after adding `actions.permissions.removeAssetPermission.completed` listener, this `asset` function parameter becomes asset or uid? So let's change it to `assetOrUid` as we already do in other places.

Just for future safety please add also this at the beginning of the function:

```js
let asset;
if (typeof assetOrUid === 'object') {
  asset = assetOrUid;
} else {
  asset = stores.selectedAsset.asset || stores.allAssets.byUid[assetOrUid];
}

// non-owner self permission removal only gives an assetUid string, not
// an object; for consistency we make it an object here
if (!asset) {
 asset = {uid: assetOrUid};
}
```

That way we always get `asset` object for the rest of the code.